inject/src/main/java/io/micronaut/context/BeanContext.java

```diff
@@ -56,6 +56,18 @@
      */
     @NonNull BeanContextConfiguration getContextConfiguration();
 
+    /**
+     * Obtain an {@link io.micronaut.context.event.ApplicationEventPublisher} for the given even type.
+     * @param eventType The event type
+     * @param <E> The event generic type
+     * @return The event publisher, never {@code null}
+     */
+    @SuppressWarnings("unchecked")
+    default @NonNull <E> ApplicationEventPublisher<E> getEventPublisher(@NonNull Class<E> eventType) {
+        Objects.requireNonNull(eventType, "Event type cannot be null");
+        return getBean(Argument.of(ApplicationEventPublisher.class, eventType));
```

Comment:
Correct way is to use:
```
 beanContext.getBeansOfType(ApplicationEventListener.class, Qualifiers.byTypeArguments(eventType.getType()))
```
In the same way as in `ApplicationEventPublisherFactory`, if I remember correctly `Argument.of(ApplicationEventPublisher.class, eventType)` would not trigger super class event listeners.

Reply:
the call to `getBean` ends up calling `ApplicationEventPublisherFactory` so I don't think anything needs to be done here